ddl/partition.go

```diff
@@ -154,3 +155,68 @@ func validRangePartitionType(col *table.Column) bool {
 		return false
 	}
 }
+
+// checkDropTablePartition checks if the partition exists and does not allow deleting the last existing partition in the table.
+func checkDropTablePartition(meta *model.TableInfo, partName string) error {
+	oldDefs := meta.Partition.Definitions
+	for _, def := range oldDefs {
+		if strings.EqualFold(def.Name.O, partName) {
+			if len(oldDefs) == 1 {
+				return errors.Trace(ErrDropLastPartition)
+			}
+			return nil
+		}
+	}
+	return errors.Trace(ErrDropPartitionNonExistent.GenByArgs(partName))
+}
+
+func removePartitionInfo(job *model.Job, t *meta.Meta, tblInfo *model.TableInfo, partName string) error {
+	oldDefs := tblInfo.Partition.Definitions
+	newDefs := make([]model.PartitionDefinition, 0, len(oldDefs)-1)
```

Comment:
why minus 1?

Reply:
@zhexuany Only one partition can be deleted per `ddl job`.

Reply:
Ah. Is it better to add "one partition can be deleted per ddl job." as a comment?